### PR TITLE
[Main Map] Add icons to map type dialog

### LINKED ISSUE #1544: [Main Map] Add map type icons for Road, Terrain, Satellite
* Use something similar to Google Maps icons

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1544 

Things done:
* Adds new icons
* Improve layout padding
* Fixes bug where the button click doesn't work after resuming activity

@JSunde @gino-m  PTAL?

### CODE REVIEW

**Review by JSunde — APPROVED** [review]

File: ground/src/main/res/layout/map_type_dialog_item.xml [hunk 1]
```diff
@@ -25,15 +25,18 @@
 
   <ImageView
     android:id="@+id/imageview"
-    android:layout_width="wrap_content"
-    android:layout_height="wrap_content"
-    tools:src="@drawable/ground_logo" />
+    android:layout_width="128dp"
+    android:layout_height="128dp"
+    android:padding="7dp"
+    tools:src="@drawable/ic_type_roadmap" />
```
Comment (gino-m) [inline ground/src/main/res/layout/map_type_dialog_item.xml:31]: Layout nit: Can we align the left and right sides of the images with the "Map type" header and close button?

File: ground/src/main/res/layout/map_type_dialog_item.xml [hunk 2]
```diff
@@ -25,15 +25,18 @@
 
   <ImageView
     android:id="@+id/imageview"
-    android:layout_width="wrap_content"
-    android:layout_height="wrap_content"
-    tools:src="@drawable/ground_logo" />
+    android:layout_width="128dp"
```
Comment (gino-m) [inline ground/src/main/res/layout/map_type_dialog_item.xml:28]: Is it easy to also give the images rounded corners?

**Review by gino-m — CHANGES_REQUESTED** [review]

**Comment by shobhitagarwal1612** [comment]
@gino-m Gentle ping!

File: ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt [hunk 3]
```diff
@@ -36,14 +37,17 @@ import timber.log.Timber
...
+  private val mapStateRepository: MapStateRepository,
+  private val settingsManager: SettingsManager,
+  private val permissionsManager: PermissionsManager,
+  mapController: MapController
 ) : AbstractViewModel() {
 
   val locationLock: MutableStateFlow<Result<Boolean>> =
     MutableStateFlow(Result.success(mapStateRepository.isLocationLockEnabled))
   private val locationLockEnabled: @Hot(replays = true) MutableLiveData<Boolean> = MutableLiveData()
+  val basemapType: LiveData<Int>
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt:50]: Nit: `baseMapType`

**Review by gino-m — APPROVED** [review]
🗺️ 👍

**Pull request merged**